Cargo.toml

```diff
@@ -41,7 +41,7 @@ pathdiff = "0.2"
 serde_derive = "1.0.8"
 float-cmp = "0.9"
 chrono = { version = "0.4", features = ["serde"] }
-tokio = { version = "1", features = ["rt-multi-thread", "macros", "fs", "io-util", "time"]}
+tokio = { version = "1.29", features = ["rt-multi-thread", "macros", "fs", "io-util", "time"]}
```

Comment:
Is this _really_ necessary? I wouldn't mind this change at all on master, but I am way more conservative with patchreleases.

I wouldn't argue too hard here, just want to make sure that you really think it is necessary here...

Reply:
TBH that was from my attempts at the other approach.  I'm trying it without.

Reply:
Awesome, thank you!

Reply:
I think this is going to work.  Should I rebase this branch without that commit?  I then need to rerun the generation script which, in practice, ends up with a `Cargo.lock` using `tokio` 1.28 rather than 1.29, so the tip will be different.

Alternatively I could push a revert of the tokio dep change, and leave the lockfile alone.

Reply:
As I expected my test passed.  (https://github.com/ijackson/config-rs/actions/runs/6629032741, but that branch is ... rather odd.)  So we should drop the change to `tokio` in `Cargo.toml`.

Should we (1) ship the `Cargo.lock` from this MR which was made *with* the `tokio` change using the recipe in the commit message (ie, `git revert` the `tokio` dependency commit), or (2) ship a `Cargo.lock` which was made *without* the `tokio` change, for somewhat greater coherency?

If 1, feel free to push that revert yourself.  It should pass CI and we can merge this.

If 2, LMK and I'll prepare the branch and force push.  (Or you can do it, e.g. to verify my work....)

Reply:
If I'm rebasing I'll fix the commit message typos @polarathene spotted.  If we're doing the "revert" approach, I could still do that.  Please let me know.

FTAOD I think either of the approaches above are fine and there is little to choose between them.

Reply:
Just to emphasize, if you don't revert this change (_or use a min version of `1.27` instead_), expect the fragile `tokio-util` pinning requirement.

Alternatively, raising `warp` will remove the `0.6` series of `tokio-util`, making it easier to pin `tokio-util` without a version ref.
- `0.7.8` is compatible with Rust `1.56.0` and will resolve to that implicitly if `tokio` is < `1.28`. Otherwise it requires Rust `1.60`.
- Or you can take the only pin some deps to their minimal versions approach instead that's been proposed. Even though that does not accurately represent using the latest compatible version possible for bugfixes / security patches that users on the MSRV likely would want and tests should be intended for ensuring is valid? 🤷‍♂️